fdb-extensions/src/test/java/com/apple/foundationdb/tuple/TupleOrderingTest.java

```diff
@@ -0,0 +1,102 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.tuple;
+
+import com.google.common.collect.Lists;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+
+import java.util.List;
+import java.util.TreeMap;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
+ * Tests of {@link TupleOrdering}.
+ */
+class TupleOrderingTest {
+
+    private static final byte[][] INVERT_BYTES = {
+        new byte[] { }, new byte[] { (byte)0x80 },
+        new byte[] { 0x00 }, new byte[] { 0x7F, 0x7F, (byte)0x86 },
+        new byte[] { 0x01 }, new byte[] { 0x7F, 0x3F, (byte)0x86 },
+        new byte[] { 0x01, 0x02 }, new byte[] { 0x7F, 0x3F, 0x3F, (byte)0x85 },
+        new byte[] { 0x01, 0x02, 0x00 }, new byte[] { 0x7F, 0x3F, 0x3F, 0x7F, (byte)0x84 },
+        new byte[] { 0x01, 0x02, 0x03 }, new byte[] { 0x7F, 0x3F, 0x3F, 0x4F, (byte)0x84 },
+        new byte[] { 0x01, 0x02, (byte)0xFF }, new byte[] { 0x7F, 0x3F, 0x20, 0x0F, (byte)0x84 },
+        new byte[] { 0x01, 0x02, (byte)0xFF, 0x00 }, new byte[] { 0x7F, 0x3F, 0x20, 0x0F, 0x7F, (byte)0x83 },
+        new byte[] { 0x01, 0x02, (byte)0xFF, 0x01 }, new byte[] { 0x7F, 0x3F, 0x20, 0x0F, 0x77, (byte)0x83 },
+        new byte[] { (byte)0xFF }, new byte[] { 0x00, 0x3F, (byte)0x86 },
+    };
+
+    @Test
+    void testInvert() {
+        for (int i = 0; i < INVERT_BYTES.length; i += 2) {
+            final byte[] original = INVERT_BYTES[i];
+            final byte[] inverted = TupleOrdering.invert(original);
+            final byte[] expected = INVERT_BYTES[i + 1];
+            assertArrayEquals(expected, inverted,
+                    () -> ByteArrayUtil2.toHexString(inverted) + " != " + ByteArrayUtil2.toHexString(expected) + " for " + ByteArrayUtil2.toHexString(original));
+        }
+    }
+
+    @Test
+    void testUninvert() {
+        for (int i = 0; i < INVERT_BYTES.length; i += 2) {
+            final byte[] inverted = INVERT_BYTES[i + 1];
+            final byte[] uninverted = TupleOrdering.uninvert(inverted);
+            final byte[] expected = INVERT_BYTES[i];
+            assertArrayEquals(expected, uninverted,
+                    () -> ByteArrayUtil2.toHexString(uninverted) + " != " + ByteArrayUtil2.toHexString(expected) + " for " + ByteArrayUtil2.toHexString(inverted));
+        }
+    }
+
+    private static final Object[] INVERT_OBJS = {
+        null,
+        new byte[] { }, new byte[] { 0x01 }, new byte[] { 0x01, 0x02 }, new byte[] { (byte)0xFF },
+        "", "a", "ab", "abc", "abcd", "abcde", "abcdef", "abcdefg", "ac",
+        Long.MIN_VALUE, -2, -1, 0, 1, 2, Long.MAX_VALUE,
+        -1e10, -2.0, -1.0, 0.0, 2.0, 1e10,
```

Comment:
In theory, this could include `-Infinity`, `-0.0`, `Infinity`, and a few positive and negative NaN values, just to cover bases. None of those should be particularly difficult cases for this algorithm, though.

Reply:
Added.